src/vm/threadstatics.cpp

```diff
@@ -500,34 +528,74 @@ void    ThreadLocalModule::AllocateDynamicClass(MethodTable *pMT)
     // We need this check because maybe a class had a cctor but no statics
     if (dwStaticBytes > 0 || dwNumHandleStatics > 0)
     {
-        // Collectible types do not support static fields yet
-        if (pMT->Collectible())
-            COMPlusThrow(kNotSupportedException, W("NotSupported_CollectibleNotYet"));
-
         if (pDynamicStatics == NULL)
         {            
+            SIZE_T dynamicEntrySize;
+            if (pMT->Collectible())
+            {
+                dynamicEntrySize = sizeof(CollectibleDynamicEntry);
+            }
+            else
+            {
+                dynamicEntrySize = DynamicEntry::GetOffsetOfDataBlob() + dwStaticBytes;
+            }
+
             // If this allocation fails, we will throw
-            pDynamicStatics = (DynamicEntry*)new BYTE[sizeof(DynamicEntry) + dwStaticBytes];
+            pDynamicStatics = (DynamicEntry*)new BYTE[dynamicEntrySize];
 
 #ifdef FEATURE_64BIT_ALIGNMENT
             // The memory block has be aligned at MAX_PRIMITIVE_FIELD_SIZE to guarantee alignment of statics
-            static_assert_no_msg(sizeof(DynamicEntry) % MAX_PRIMITIVE_FIELD_SIZE == 0);
+            static_assert_no_msg(sizeof(NormalDynamicEntry) % MAX_PRIMITIVE_FIELD_SIZE == 0);
             _ASSERTE(IS_ALIGNED(pDynamicStatics, MAX_PRIMITIVE_FIELD_SIZE));
 #endif
 
             // Zero out the new DynamicEntry
-            memset((BYTE*)pDynamicStatics, 0, sizeof(DynamicEntry) + dwStaticBytes);
+            memset((BYTE*)pDynamicStatics, 0, dynamicEntrySize);
+
+            if (pMT->Collectible())
+            {
+                ((CollectibleDynamicEntry*)pDynamicStatics)->m_pLoaderAllocator = pMT->GetLoaderAllocator();
+            }
 
             // Save the DynamicEntry in the DynamicClassTable
             m_pDynamicClassTable[dwID].m_pDynamicEntry = pDynamicStatics;
         }
 
+        if (pMT->Collectible() && (dwStaticBytes != 0))
+        {
+            GCX_COOP();
```

Comment:
>GCX_COOP(); [](start = 12, length = 11)

This function is already MODE_COOPERATIVE, it shouldn't need GCX_COOP()

Reply:
Right, removing it.